### ISSUE TITLE
Explicitly add GPLv3-or-later to license list

### ISSUE THREAD

**Issue by lofidevops** [go-gitea/gitea#2842]
- Gitea version (or commit ref): 2406094
- Git version: N/A
- Operating system: N/A
- Database (use `[x]`): N/A
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL) https://try.gitea.io/repo/create
- Log gist: N/A

## Description

The GNU ethical repository criteria (see #1524) suggests that a host "recommends and encourages GPL 3-or-later licensing at least as much as any other kind of licensing"

At the moment the license dropdown simply lists "GPL-3.0" and places the GPL-3.0 license text in the repository. Unfortunately this is ambiguous and could be interpreted as GPL-3.0-only rather than GPL-3.0-or-later.

## Suggested solution 1

1. Modify the dropdown list as follows:

* AGPL-3.0-only
* AGPL-3.0-or-later
* GPL-2.0-only
* GPL-2.0-or-later
* GPL-3.0-only
* GPL-3.0-or-later

(I see GPL-1.0 is also listed, I'm not sure how this applies to that license.)

2. Always include the full license text (current behaviour)

3. When populating boilerplate paragraphs include the "or (at your option) any later version" text as appropriate.

## Suggested solution 2

As above, but only list the "-or-later" options. Anyone who wants "-only" can do their licensing manually.

**Comment by bill-auger** [comment]
this was just discussed on the mailing list - the premise of this issue is probably mis-guided - as i understand, all that the license selector does is add a verbatim copy of the GPL to the repo - there is nothing in the GPL itself that determines "or any later version" or not - so even if these were separate options on the GUI, the actual behavior would necessarily be identical - there is only one GPLv3 and it should never be modified

the declaration of "or any later version" appears only in the header of the source code files which at the time of repo creation do not yet exist - im not sure what are the "boilerplate paragraphs" mentioned here but they would be an inadequate carrier of that information in 99% of cases

**Comment by bkcsoft** [comment]
We don't recommend any licence what so ever (hence they are equally unrecommended). They are listed in alphabetical order and they are vendored from https://github.com/spdx/license-list where the names also comes from. 

As for `3.0-or-later`, until FSF releases a license _with that name_ I'm going to read that as "GPL v3, or any later version of GPL", which GPLv3 falls under.

**Comment by lofidevops** [comment]
Thanks for the feedback, I stand corrected. My understanding is that the SPDX may be updated to list both `GPL-3.0-only` and `GPL-3.0+` but this is an upstream issue (and still under discussion).

**Comment by bill-auger** [comment]
it is sensible for SPDX to add them (GPL-3.0-only and GPL-3.0+) as distinct use-cases, but this is meaningful only after the source code is published - neither gitea nor SPDX can do anything to declare this distinction at the time of repo creation on the website - it would be absolutely pointless to present more than one GPLv2 and  one GPLv3 select options on the GUI - there is only one single possible action that gitea could take for both cases - that is to add THE (one and only) GPLv3 to the repo - the distinction of "or any later version" can only be specified explicitly by the author of the project source code and this is typically done in the copyright header of project source code files which do not exist at the time of the website repo creation

**Comment by lofidevops** [comment]
What if Gitea did something like:

Given user has picked a license
And user is informed a readme template must be selected
And user picks a readme template (otherwise default is used)
When repo is created
Then license text is added to repo
And license summary is added to readme

(just a rough idea, happy to create a new ticket to hash it out)

**Comment by bill-auger** [comment]
you could do that but it is very unconventional - the README file rarely contains any licensing information - the text of the GPL itself recommends stating "or any later version" explicitly in each and every source code file and that is generally what everyone does

if the intention here is to make the license more visible to on-lookers then i would suggest a license detector script would be more helpful that shows the detected license at the top of the webpage like github does

**Comment by bkcsoft** [comment]
@kwill Gitea will never _force_ the user to do anything, esp not to pick a license.

@bill-auger Most projects have a license header at the bottom though 🤔
